forester/src/indexer/photon_indexer.rs

```diff
@@ -1,28 +1,32 @@
 use crate::utils::decode_hash;
 use account_compression::initialize_address_merkle_tree::Pubkey;
 use light_test_utils::indexer::{Indexer, IndexerError, MerkleProof, NewAddressProofWithContext};
-use log::info;
+use light_test_utils::rpc::rpc_connection::RpcConnection;
+use log::{debug, info};
 use photon_api::apis::configuration::Configuration;
 use photon_api::models::GetCompressedAccountsByOwnerPostRequestParams;
 use solana_sdk::bs58;
 use std::fmt::Debug;
 
-pub struct PhotonIndexer {
+pub struct PhotonIndexer<const INDEXED_ARRAY_SIZE: usize, R: RpcConnection> {
```

Comment:
Const generics here feel weird to me since the local photon client will not hold an indexed array.

Reply:
Agreed, it's ugly. Test indexer has generic R in one of the methods, that needs to be unified in trait. Here: 
1. https://github.com/Lightprotocol/light-protocol/pull/914/files#diff-f00f25b99320fcd210d228457a61aad543f557fb69cfca88889f455db0d396c2R105
2. https://github.com/Lightprotocol/light-protocol/pull/914/files#diff-deb92dc32bc5ec1000b929e1de32f88bc0f510331eca1f276dc4fe450da7db37R498-R505

Reply:
I think we should just remove the const generic from indexed array in the first place.

I must've been braindead when introducing it in the first place. It's not used on-chain, so there is absolutely no reason to use arrays. We could probably just use a simple `Vec` there.

Reply:
I think we did use it onchain for a little bit before we had hashset queues